src/main/java/teammates/ui/webapi/action/GetInstructorsAction.java

```diff
@@ -76,7 +77,18 @@ public ActionResult execute() {
             });
         } else if (intentStr.equals(Intent.FULL_DETAIL.toString())) {
             // get all instructors of a course without information hiding
-            data = new InstructorsData(instructorsOfCourse);
+            // adds googleId if caller is course co-owner or admin
+            if (userInfo.isAdmin || logic.getInstructorForGoogleId(courseId, userInfo.getId()).getRole().equals(
+                    Const.InstructorPermissionRoleNames.INSTRUCTOR_PERMISSION_ROLE_COOWNER)) {
```

Comment:
I have checked the access control logic again. It is not necessarily only co-owner can see the Google ID, but rather anyone who has `canmodifyinstructor` permission.